runner.go

```diff
@@ -230,7 +230,7 @@ func (runner *Runner) StartWorker(id string, startFunc func() (Worker, error)) e
 		// the startc arm of the select, and that calls startWorker (which never blocks)
 		// and then immediately sends any error to the reply channel.
 		return <-reply
-	case <-runner.tomb.Dead():
+	case <-runner.tomb.Dying():
```

Comment:
I think this needs a new case for `Dying` with a different return. This causes us to return `ErrDead` on `Dying`, which isn't quite right.

Reply:
Yes you're right I'll do it when I get back